Source/Internal/IGListDebugger.m

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "IGListDebugger.h"
+
+#import "IGListDebuggingUtilities.h"
+#import "IGListAdapter+DebugDescription.h"
+
+@implementation IGListDebugger
+
+static NSHashTable<IGListAdapter *> *livingAdaptersTable = nil;
+
++ (void)trackAdapter:(IGListAdapter *)adapter {
+#if IGLK_DEBUG_DESCRIPTION_ENABLED
+    if (livingAdaptersTable == nil) {
+        livingAdaptersTable = [NSHashTable weakObjectsHashTable];
+    }
+    [livingAdaptersTable addObject:adapter];
```

Comment:
Are `IGListAdapter` instances only created on main thread? For future consideration: an alternative to tracking using weak references (which involves a global lock and I believe may cause the objc runtime to use slow path in some cases), you could track `__unsafe_unretained` pointers and have them be removed them from the hash table from `-[IGListAdapter dealloc]`.